pkg/api/client.go

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+type CertificateResponse struct {
+	CertPEM  []byte
+	ChainPEM []byte
+	SCT      []byte
+}
+
+// SigstorePublicServerURL is the URL of Sigstore's public Fulcio service.
+const SigstorePublicServerURL = "https://fulcio.sigstore.dev"
+
+// Client is the interface for accessing the Fulcio API.
+type Client interface {
+	// SigningCert sends the provided CertificateRequest to the /api/v1/singingCert
+	// endpoint of a Fulcio API, authenticated with the provided bearer token.
+	SigningCert(cr CertificateRequest, token string) (*CertificateResponse, error)
+}
+
+// NewClient creates a new Fulcio API client talking to the provided URL.
+func NewClient(url *url.URL) Client {
```

Comment:
could be an additional patch, but would be nice to be able to override the client here (e.g. to set a specific timeout value)

Reply:
Let's follow up with that, I opened: https://github.com/sigstore/fulcio/issues/266